src/pcsclite.cpp

```
@@ -180,17 +184,18 @@ void PCSCLite::HandlerFunction(void* arg) {
                                           INFINITE,
                                           &pcsclite->m_card_reader_state,
                                           1);
+
             uv_mutex_lock(&pcsclite->m_mutex);
             if (pcsclite->m_state) {
                 uv_cond_signal(&pcsclite->m_cond);
-            }
-
-            if (result != SCARD_S_SUCCESS) {
+            } else if (result != SCARD_S_SUCCESS) {
                 pcsclite->m_state = 2;
             }
-
             uv_mutex_unlock(&pcsclite->m_mutex);
         } else {
+            // Really need to exit if previous result is not success ?
```

I think we should return an error  and exit